src/gt4py/analysis/passes.py

```diff
@@ -700,6 +706,28 @@ def visit_TernaryOpExpr(self, node: gt_ir.TernaryOpExpr, **kwargs):
                 node.then_expr.data_type, node.else_expr.data_type
             )
 
+        def visit_NativeFuncCall(self, node: gt_ir.NativeFuncCall, **kwargs):
+            self.generic_visit(node.args, **kwargs)
+
+            # Collect data types from each argument
+            data_type = set([arg.data_type for arg in node.args])
+            if len(data_type) > 1:
+                data_type -= {gt_ir.DataType.DEFAULT}
+
+            # Assert that all data types are the same
+            assert len(data_type) == 1
+            data_type = data_type.pop()
+
+            if node.func in (
+                gt_ir.NativeFunction.MIN,
+                gt_ir.NativeFunction.MAX,
+                gt_ir.NativeFunction.ABS,
+                gt_ir.NativeFunction.MOD,
+            ):
+                node.data_type = data_type
+            else:
+                node.data_type = gt_ir.DataType.DEFAULT
```

Comment:
Why is this correct? I mean, why should the result all of the other builtin functions be a float/double? What about  boolean functions like `isfinite`, `isinf`,`isnan` or other argument dependant ones like `floor` and `ceil`?

Reply:
good catch